src/manager.rs

```diff
@@ -214,6 +231,7 @@ impl KeyHolderWrite {
             }
             None => {
                 self.key.delete()?;
+                self.val = None;
```

Comment:
Should we clear the cached value before calling `delete()`?
In case a keyspace notification would trigger an API call that might use the cached value although it was deleted?
The disadvantage of clearing the cache first, is that if delete fails, the cache was cleared in vain, but this is less severe in my opinion, than caching a stale value.

Reply:
Good point :+1:

Reply:
Though if a key space notification will cause opening the key then it will be another write holder with its own cache. But I agree its safer this way...